dbms/src/Storages/StorageMergeTree.cpp

```diff
@@ -261,6 +261,15 @@ void StorageMergeTree::alter(
 
     if (primary_key_is_modified)
         data.loadDataParts(false);
+
+    /// Do freeze of all parts after all other operations.
```

Comment:
What about StorageReplicatedMergeTree?

Reply:
Added the same code to `StorageReplicatedMergeTree`